### PR TITLE
Query on slice state

### LINKED ISSUE #70: Querying certificates returns slices that are not available

### PR DESCRIPTION
**Pull request by duizer**
Fixes #70

### CODE REVIEW

File: src/ProjectOrigin.WalletSystem.IntegrationTests/TransferCertificateTests.cs [hunk 1]
```diff
@@ -64,6 +64,49 @@ public async Task Transfer_SingleSlice_LocalWallet()
...
+        var request = new TransferRequest()
+        {
+            CertificateId = issuedEvent.CertificateId,
+            Quantity = transferredAmount,
+            ReceiverId = receiverEndpointResponse.ReceiverId
+        };
+        await client.TransferCertificateAsync(request, senderHeader);
+
+        //Assert
+        await WaitForCertCount(client, recipientHeader, 1);
```
Comment (ckr123) [inline src/ProjectOrigin.WalletSystem.IntegrationTests/TransferCertificateTests.cs:100]: I am not sure what this method does and why it is needed. Could the name be more specific?

File: src/ProjectOrigin.WalletSystem.Server/Repositories/CertificateRepository.cs [hunk 2]
```diff
@@ -76,8 +76,8 @@ public async Task<IEnumerable<CertificateViewModel>> GetAllOwnedCertificates(str
                     JOIN Slices s ON de.Id = s.DepositEndpointId
                     JOIN Certificates c ON s.CertificateId = c.Id
                     LEFT JOIN Attributes a ON c.Id = a.CertificateId AND c.RegistryId = a.RegistryId
-                    JOIN Registries r ON c.RegistryId = r.Id
-                    WHERE w.Owner = @owner";
+                    JOIN Registries r ON c.RegistryId = r.Id 
+                    WHERE w.Owner = @owner AND s.SliceState = 1";
```
Comment (ckr123) [inline src/ProjectOrigin.WalletSystem.Server/Repositories/CertificateRepository.cs:80]: What is SliceState = 1?

Comment (duizer) [reply src/ProjectOrigin.WalletSystem.Server/Repositories/CertificateRepository.cs:80]: It means that it is available. SliceState is an enum value:

```cs
public enum SliceState
{
    Available = 1,
    Slicing = 2, // Reserved
    Registering = 3,
    Sliced = 4,
    Transferred = 5,
}
```

Comment (duizer) [reply src/ProjectOrigin.WalletSystem.Server/Repositories/CertificateRepository.cs]: Please see the changes in 21e120f5463f4b312f8ede728dd24bc0294cdb5f. That should be easier to read.

File: src/ProjectOrigin.WalletSystem.IntegrationTests/TransferCertificateTests.cs (hunk 1 above)
Comment (duizer) [reply src/ProjectOrigin.WalletSystem.IntegrationTests/TransferCertificateTests.cs]: I have deleted this test and added a new one in the test of the certificates repository.

File: src/ProjectOrigin.WalletSystem.IntegrationTests/Repositories/CertificateRepositoryTests.cs [hunk 3]
```diff
@@ -123,6 +123,32 @@ public async Task GetAllOwnedCertificates()
         );
     }
 
+    [Theory]
+    [InlineData(SliceState.Registering)]
+    [InlineData(SliceState.Sliced)]
+    [InlineData(SliceState.Slicing)]
+    [InlineData(SliceState.Transferred)]
```
Comment (MartinSchmidt) [inline src/ProjectOrigin.WalletSystem.IntegrationTests/Repositories/CertificateRepositoryTests.cs:130]: Could add available as well and either the count or boolean flag if any result should be found, then all scenarios is tested

**Review by MartinSchmidt — APPROVED** [review]

File: src/ProjectOrigin.WalletSystem.IntegrationTests/Repositories/CertificateRepositoryTests.cs (hunk 3 above)
Comment (MartinSchmidt) [reply src/ProjectOrigin.WalletSystem.IntegrationTests/Repositories/CertificateRepositoryTests.cs:130]: @duizer  but is up to you

Comment (duizer) [reply src/ProjectOrigin.WalletSystem.IntegrationTests/Repositories/CertificateRepositoryTests.cs]: I like it. I have it changed it.

**Pull request merged**